Benchmarks fail to build for 0.2.0.0 on Stackage Nightly

It appears that the benchmarks fail to build for `fused-effects-0.2.0.0` with the errors in the snippet below. 

Unless there are any objections, I'm going to disable the benchmarks on Stackage's nightly build for the time being; I'll link this issue to that commit so it's easy to keep track of.

<details><summary>Error Snippet</summary>
<p>

```
[1 of 1] Compiling Main             ( benchmark/Bench.hs, dist/build/benchmark/benchmark-tmp/Main.o )

benchmark/Bench.hs:15:41: error:
    • Couldn't match type ‘Sum Int’ with ‘Eff VoidC’
      Expected type: Int -> Eff VoidC b0
        Actual type: Int -> Sum Int b0
    • In the second argument of ‘(.)’, namely
        ‘execWriter @_ @_ @(Sum Int) . tellLoop’
      In the first argument of ‘whnf’, namely
        ‘(run . execWriter @_ @_ @(Sum Int) . tellLoop)’
      In the second argument of ‘($)’, namely
        ‘whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100’
   |
15 |       [ bench "100"       $ whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

benchmark/Bench.hs:15:60: error:
        Actual type: Int -> Sum Int b1
    • In the second argument of ‘(.)’, namely
        ‘execWriter @_ @_ @(Sum Int) . tellLoop’
      In the first argument of ‘whnf’, namely
        ‘(run . execWriter @_ @_ @(Sum Int) . tellLoop)’
      In the second argument of ‘($)’, namely
        ‘whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100000’
   |
16 |       , bench "100000"    $ whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100000
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

benchmark/Bench.hs:16:60: error:
    • Expected kind ‘* -> *’, but ‘Sum Int’ has kind ‘*’
    • In the type ‘(Sum Int)’
      In the first argument of ‘(.)’, namely
        ‘execWriter @_ @_ @(Sum Int)’
      In the second argument of ‘(.)’, namely
        ‘execWriter @_ @_ @(Sum Int) . tellLoop’
   |
16 |       , bench "100000"    $ whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100000
   |                                                            ^^^^^^^

benchmark/Bench.hs:17:41: error:
    • Couldn't match type ‘Sum Int’ with ‘Eff VoidC’
      Expected type: Int -> Eff VoidC b2
        Actual type: Int -> Sum Int b2
    • In the second argument of ‘(.)’, namely
        ‘execWriter @_ @_ @(Sum Int) . tellLoop’
      In the first argument of ‘whnf’, namely
        ‘(run . execWriter @_ @_ @(Sum Int) . tellLoop)’
      In the second argument of ‘($)’, namely
        ‘whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100000000’
   |
17 |       , bench "100000000" $ whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100000000
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

benchmark/Bench.hs:17:60: error:
    • Expected kind ‘* -> *’, but ‘Sum Int’ has kind ‘*’
    • In the type ‘(Sum Int)’
      In the first argument of ‘(.)’, namely
        ‘execWriter @_ @_ @(Sum Int)’
      In the second argument of ‘(.)’, namely
        ‘execWriter @_ @_ @(Sum Int) . tellLoop’
   |
17 |       , bench "100000000" $ whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100000000
   |   
```

</p>
</details>

--- a/.travis.yml
+++ b/.travis.yml
@@ -33,7 +33,7 @@ before_install:
 
 install:
 - cabal new-update
-- cabal new-configure --enable-tests --write-ghc-environment-files=always
+- cabal new-configure --enable-benchmarks --enable-tests --write-ghc-environment-files=always
 - cabal new-build --only-dependencies
 
 script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -38,6 +38,7 @@ install:
 
 script:
 - cabal new-build
+- cabal new-build benchmark
 - cabal new-test
 - cabal new-haddock
 - cabal new-sdist

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -1,3 +1,7 @@
+# 0.2.0.1
+
+- Fixes the benchmarks, and builds them in CI to avoid regressing them again.
+
 # 0.2.0.0
 
 - Adds `listen`, `listens`, and `censor` operations to `Writer`.

--- a/benchmark/Bench.hs
+++ b/benchmark/Bench.hs
@@ -12,9 +12,9 @@ main :: IO ()
 main = defaultMain
   [ bgroup "WriterC"
     [ bgroup "Eff"
-      [ bench "100"       $ whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100
-      , bench "100000"    $ whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100000
-      , bench "100000000" $ whnf (run . execWriter @_ @_ @(Sum Int) . tellLoop) 100000000
+      [ bench "100"       $ whnf (run . execWriter @(Sum Int) . tellLoop) 100
+      , bench "100000"    $ whnf (run . execWriter @(Sum Int) . tellLoop) 100000
+      , bench "100000000" $ whnf (run . execWriter @(Sum Int) . tellLoop) 100000000
       ]
     , bgroup "standalone"
       [ bench "100"       $ whnf (fst . runVoidC . flip runWriterC (Sum (0 :: Int)) . tellLoop) 100

--- a/fused-effects.cabal
+++ b/fused-effects.cabal
@@ -1,5 +1,5 @@
 name:                fused-effects
-version:             0.2.0.0
+version:             0.2.0.1
 synopsis:            A fast, flexible, fused effect system.
 description:         A fast, flexible, fused effect system, à la Effect Handlers in Scope, Monad Transformers and Modular Algebraic Effects: What Binds Them Together, and Fusion for Free—Efficient Algebraic Effect Handlers.
 homepage:            https://github.com/robrix/fused-effects

